nats-spring/src/main/java/io/nats/spring/boot/autoconfigure/NatsConnectionProperties.java

```diff
@@ -656,58 +785,95 @@ protected KeyStore loadKeystore(String path, char[] password) throws IOException
+			throws IOException, GeneralSecurityException {
+		if (keyStoreProvider == null || keyStoreProvider.length() == 0) {
+			keyStoreProvider = defaultKeyStoreProviderAlgorithm;
+		}
+
+		if (keyStoreType == null || keyStoreType.length() == 0) {
+			keyStoreType = defaultKeyStoreType;
 		}
 
 		if (password == null || password.length == 0) {
 			password = new char[0];
 		}
 
-		KeyStore store = this.loadKeystore(path, password);
-		KeyManagerFactory factory = KeyManagerFactory.getInstance(type);
+		KeyStore store = this.loadKeystore(path, password, keyStoreType);
+		KeyManagerFactory factory = KeyManagerFactory.getInstance(keyStoreProvider);
 		factory.init(store, password);
 		return factory.getKeyManagers();
 	}
 
-	protected TrustManager[] createTrustManagers(String path, char[] password, String type) throws IOException, GeneralSecurityException {
-		if (type == null || type.length() == 0) {
-			type = "SunX509";
+	protected TrustManager[] createTrustManagers(String path, char[] password, String trustStoreProvider,
+			String trustStoreType) throws IOException, GeneralSecurityException {
+		if (trustStoreProvider == null || trustStoreProvider.length() == 0) {
+			trustStoreProvider = defaultTrustStoreProviderAlgorithm;
+		}
+
+		if (trustStoreType == null || trustStoreType.length() == 0) {
+			trustStoreType = defaultTrustStoreType;
 		}
 
 		if (password == null || password.length == 0) {
 			password = new char[0];
 		}
 
-		KeyStore store = loadKeystore(path, password);
-		TrustManagerFactory factory = TrustManagerFactory.getInstance(type);
-		factory.init(store);
+		KeyStore store = loadKeystore(path, password, trustStoreType);
+		TrustManagerFactory factory = TrustManagerFactory.getInstance(trustStoreProvider);
+		
+		/* 
+		 * Trust Store doesn't need to load any keys from the store (just certs)
+		 * so passing a password is not necessary
+		 */
+		factory.init(store); 
+		
 		return factory.getTrustManagers();
 	}
 
-	protected SSLContext createSSLContext() throws IOException, GeneralSecurityException  {
-		SSLContext ctx = SSLContext.getInstance(Options.DEFAULT_SSL_PROTOCOL);
-		ctx.init(this.createKeyManagers(this.keyStorePath, this.keyStorePassword, this.keyStoreType),
-					this.createTrustManagers(this.trustStorePath, this.trustStorePassword, this.trustStoreType), new SecureRandom());
+	/**
+	 * Create a SSLContext with the given tlsProtocol
+	 * 
+	 * @return SSLContext
+	 * @throws IOException
+	 * @throws GeneralSecurityException
+	 */
+	protected SSLContext createSSLContext() throws IOException, GeneralSecurityException {
+		
+		if (this.tlsProtocol == null || this.tlsProtocol.length() == 0) {
+			this.tlsProtocol = Options.DEFAULT_SSL_PROTOCOL;
+		}
+
+		SSLContext ctx = SSLContext.getInstance(this.tlsProtocol);
+		ctx.init(
+				this.createKeyManagers(this.keyStorePath, this.keyStorePassword, this.keyStoreProvider,
```

Comment:
I think the formatting is broken here :/ This change makes the code harder to read.
the `this.createKeyManagers(this.keyStorePath, this.keyStorePassword` should start at the index of the the bracket from before `ctx.init(` - this would give you more space to not break the line
The closing bracket from `ctx.init(` could also have its own line as the start is having.

Reply:
Ugh! The entire file was mangled, sorry about that. I blame my bad habit of mindlessly indenting/reformatting code all the time.
Thanks for reviewing it!